test/foundry/deploy/03_Erc20Enforcer.t.sol

```diff
@@ -0,0 +1,697 @@
+  }
+
+  function testSafeTransferFrom() public {
+    mintToDeployer();
+    approvalToAlice(initialValue);
+    vm.expectEmit(true, true, false, true);
+    emit Transfer(address(deployer), address(alice), initialValue);
+    vm.prank(alice);
+    holographERC20.safeTransferFrom(address(deployer), address(alice), initialValue);
+  }
+
+  function testBalanceOfDeployerAfterSafeTransferFrom() public {
+    testSafeTransferFrom();
+    assertEq(holographERC20.balanceOf(deployer), 0);
+  }
+
+  function testBalanceOfAliceAfterSafeTransferFrom() public {
+    testSafeTransferFrom();
+    assertEq(holographERC20.balanceOf(alice), initialValue);
+  }
+
+  function testTransferFrom() public {
+    mintToDeployer();
+    vm.expectEmit(true, true, false, true);
+    emit Approval(deployer, alice, initialValue);
+    approvalToAlice(initialValue);
+    //check allowance alice = 1
+    assertEq(holographERC20.allowance(deployer, alice), initialValue);
+    vm.expectEmit(true, true, false, true);
+    emit Transfer(address(deployer), address(alice), initialValue);
+    vm.prank(alice);
+    holographERC20.transferFrom(address(deployer), address(alice), initialValue);
+    //check allowance alice = 0
+    assertEq(holographERC20.allowance(deployer, alice), 0);
+  }
+
+  function testSafeTransferToErc20Reciver() public {
+    erc20Mock.toggleWorks(true);
+    mintToDeployer();
+    vm.expectEmit(true, true, false, true);
+    emit Transfer(address(deployer), address(erc20Mock), initialValue);
+    vm.prank(deployer);
+    holographERC20.safeTransfer(address(erc20Mock), initialValue);
+    vm.expectEmit(true, true, false, true);
+    emit Transfer(address(erc20Mock), address(deployer), initialValue);
+    erc20Mock.transferTokens(payable(holographERC20), deployer, initialValue);
+  }
+
+  function testSafeTransferWithBytesToErc20Reciver() public {
+    erc20Mock.toggleWorks(true);
+    mintToDeployer();
+    vm.expectEmit(true, true, false, true);
+    emit Transfer(address(deployer), address(erc20Mock), initialValue);
+    vm.prank(deployer);
+    holographERC20.safeTransfer(address(erc20Mock), initialValue, zeroBytes);
+  }
+
+  function testSafeTransferFromToErc20() public {
+    mintToDeployer();
+    erc20Mock.toggleWorks(true);
+    vm.expectEmit(true, true, false, true);
+    emit Approval(deployer, alice, initialValue);
+    approvalToAlice(initialValue);
+    //check allowance alice = 1
+    assertEq(holographERC20.allowance(deployer, alice), initialValue);
+    vm.expectEmit(true, true, false, true);
+    emit Transfer(address(deployer), address(erc20Mock), initialValue);
+    vm.prank(alice);
+    holographERC20.safeTransferFrom(address(deployer), address(erc20Mock), initialValue);
+    //check allowance alice = 0
+    assertEq(holographERC20.allowance(deployer, alice), 0);
+  }
+
+  function testSafeTransferFromBytesToErc20() public {
+    mintToDeployer();
+    erc20Mock.toggleWorks(true);
+    vm.expectEmit(true, true, false, true);
```

Comment:
question: does it make sense to check for the emission of the Approval on every call to the function approvalToAlice? I feel that these expectEmit  s before the approvalToAlice calls are not needed.

Reply:
Yes, you are absolutely right, it was already test in the previous one.

Reply:
also the assert of allowance to alice